indexing-service/src/main/java/org/apache/druid/indexing/compact/CompactionSupervisor.java

```diff
@@ -76,6 +84,11 @@ public SupervisorReport<AutoCompactionSnapshot> getStatus()
       snapshot = AutoCompactionSnapshot.builder(dataSource)
                                        .withStatus(AutoCompactionSnapshot.AutoCompactionScheduleStatus.NOT_ENABLED)
                                        .build();
+    } else if (!supervisorSpec.getValidationResult().isValid()) {
+      throw InvalidInput.exception(
+          "Compaction supervisor spec is invalid. Reason[%s].",
+          supervisorSpec.getValidationResult().getReason()
+      );
```

Comment:
Let's not throw an exception here. We can either add new fields to `AutoCompactionSnapshot` or add a new class that is returned by this method.